lib/httpMock.js

```diff
@@ -15,15 +15,13 @@ function mockTemplate() {
 
         $provide.decorator('$http', ['$delegate', '$q', '$injector', function($http, $q, $injector) {
 
-        var interceptors = $httpProvider.interceptors;
+        var interceptors = [];
 
-        function getInterceptor(interceptorExpression) {
-            if (angular.isString(interceptorExpression)) {
-                return $injector.get(interceptorExpression);
-            } else {
-                return $injector.invoke(interceptorExpression);
-            }
-        }
+        angular.forEach($httpProvider.interceptors, function(interceptorExpression) {
```

Comment:
Maybe. I can not answer this question.

However we had problems with angular-loading-bar and protractor-http-mock working together. The root cause was that a stateful interceptor by anonymous factory which caused troubles with protractor-http-mock. It is hard to blame and fix one of each side in this case.

If the interceptor uses a named provider, the $injector can cache and reuse the instance and this PR can be dropped. But this issue is hard to track down and it would be nice if it does not occur in the future.